app/scripts/controllers/permissions/index.js

```diff
@@ -267,6 +267,21 @@ export class PermissionsController {
     this._removePendingApproval(id)
   }
 
+  /**
+   * Rejects all pending permissions requests
+   * (Used in the event of closing a locked request window)
+   */
+  async rejectAllPermissionsRequests () {
+    if (!this.pendingApprovals.size) {
+      log.debug(`No permissions request in queue`)
+      return
+    }
+
+    this.pendingApprovals.forEach(async (_approval, id) => {
+      await this.rejectPermissionsRequest(id)
+    })
```

Comment:
are these async function calls guaranteed to be resolved at the end of the forEach loop? I'm wondering if they should be joined / awaited before the function returns

Reply:
Given behavior in previous code for cancellations, they are, but would you prefer to chain these in a `Promise.all` call?

Reply:
hrmm, yeah I see this pattern in other places in this file, that's fair. basically what I'm trying to avoid is the race condition I document in `asyncLog` here:

https://jsfiddle.net/jd62gw0t/7/

You could collect all the async function calls and use `Promise.all`, or just use a flat for loop and make sure the enclosing async function is `await`ed properly, which would be less overhead (see `asyncLog2` from the jsfiddle.

do you know if the code that calls into `rejectAllPermissionRequests` properly `await`s the result?